src/particle_restart.cpp

```diff
@@ -100,8 +102,26 @@ void run_particle_restart()
   }
   init_particle_seeds(particle_seed, p.seeds_);
 
+  // Force calculation of cross-sections by setting last energy to zero
+  if (settings::run_CE) {
+    for (auto& micro : p.neutron_xs_) micro.last_E = 0.0;
+  }
+
+  // Prepare to write out particle track.
+  if (p.write_track_) add_particle_track(p);
+
+  // Every particle starts with no accumulated flux derivative.
+  if (!model::active_tallies.empty())
+  {
+    p.flux_derivs_.resize(model::tally_derivs.size(), 0.0);
+    std::fill(p.flux_derivs_.begin(), p.flux_derivs_.end(), 0.0);
+  }
+  
+  // Allocate space for tally filter matches
+  p.filter_matches_.resize(model::tally_filters.size());
```

Comment:
The additions here are things that appear exactly as in `initialize_history`. Maybe we should separate them out into a function somewhere?

Reply:
Yes, that's definitely a good idea. I've split this section into a new helper function so as to avoid the repetition. I couldn't think of a good name for it, so went with "initialize_history_partial()", but maybe there is a better name out there for these tasks -- let me know if something comes to mind.